src/classes/CW_AudienceGroupCompControlerTest.cls

```diff
@@ -0,0 +1,270 @@
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+			//Tray to delete used control groups
+			compCtrl.selectedGroupIds.Add(listGroups[0].Id);
+			compCtrl.selectedGroupIds.Add(listGroups[1].Id);
+			compCtrl.selectAll = false;
+			compCtrl.selectDeselectAllItems();
+			
+			compCtrl.getDeleteMessage();
+			compCtrl.deleteGroups();
+			
+			//check for errors
+			system.assertEquals(true  ,ApexPages.hasMessages());
+			system.assertEquals(14    , RecList.Size());
+			
+			compCtrl.contextItem = listGroups[4].Id;
+			compCtrl.doSelectItem();
+			system.assertEquals(1     ,compCtrl.getSelectedCount());
+			compCtrl.doDeselectItem();
+			system.assertEquals(0     ,compCtrl.getSelectedCount());
+			
+			compCtrl.last();
+			system.assertEquals(4     ,compCtrl.campGroups.Size());
+			
+			compCtrl.first();
+			system.assertEquals(5     ,compCtrl.campGroups.Size());
+			
+			compCtrl.doNext();
+			system.assertEquals(2     ,compCtrl.getPageNumber());
+			system.assertEquals(true  ,compCtrl.getHasPrevious());
+			system.assertEquals(true  ,compCtrl.getHasNext());
+
+			
+			compCtrl.doPrevious();
+			system.assertEquals(false  ,compCtrl.getHasPrevious());
+			system.assertEquals(true   ,compCtrl.getHasNext());
+
+			compCtrl.editGroups();
+			system.assertEquals(false ,compCtrl.getOutputMode());
+			system.assertEquals(false ,compCtrl.savedByParent);
+			compCtrl.savedByParent =true;
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+			compCtrl.Cancel();
+			system.assertEquals(true  ,compCtrl.getOutputMode());
+	    }
+	}
+	static testMethod void WraperClasssControllerTest() 
+	{
+        initCustomSettings();
+        
+        CW_Campaign__c camp = TestDataUtil.createCampaign();
+        insert camp;
+        
+        List<CW_Audience_Group__c> listGroups = TestDataUtil.createAudienceGroups(camp.Id,5);
+        insert listGroups;
+        
+        User thisUser = [SELECT Id FROM User WHERE Id = :UserInfo.getUserId()];
+    	
+    	Test.startTest();
+    	
+		system.runAs(thisUser)
+	    {  
+
+			List<sObject> RecList= new List<sObject>();
+			
+			//Initialization 
+	    	CW_AudienceGroupCompControler compCtrl= new CW_AudienceGroupCompControler();
+	    	CompCtrl.campId=camp.Id;
+			compCtrl.audienceGroups=RecList;
+			compCtrl.isReadOnly=false;
+			
+			CW_AudienceGroupCompControler.audienceGroupWrapper rec=compCtrl.campGroups[1];
+			
+			system.assertEquals(0     , rec.members);
+			system.assertEquals(false , rec.getIsUsed());
+			system.assertEquals(true , rec.isControlGroup);
+			rec.isControlGroup=false;
+			rec.aGroup.Control_Group__c=listGroups[0].Id;
+			system.assertEquals(false , rec.isControlGroup);
+			rec.isControlGroup=true;
+			system.assertEquals(true , rec.isControlGroup);
+			system.assertEquals(null , rec.aGroup.Control_Group__c);
+			rec.isControlGroup=false;
+	    	system.assertEquals(listGroups[0].Id , rec.aGroup.Control_Group__c);
+	    	
+	    }
+	}
+
+	static testMethod void editSaveGroupsTest() 
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility